### DISCUSSION TITLE
Getting filename from directory list

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#5638] [Q&A] [answered]
I have the following code and am trying to read in the fileName from the ones read in and add it to the end of the array.
file.Name() and file.getName() doesn't work. Could someone help out?

```
var folder = "/folders/sftp_users/CTHCare/files/qa/";
//create a directory object
var dir = new java.io.File(folder);

//get a list of file names that end in .csv
var files = dir.list(new org.apache.commons.io.filefilter.SuffixFileFilter(".csv"));

// loop through the file list
for each(file in files)
{
                //read in the contents of the file and split it on carriage return
                var arrContents = FileUtil.read(folder + file).toString().split('\n');
                logger.info('filename ' + file);
                
                //arrContents.push(file.name());
                
                //Return the delimited values
                return arrContents[1];
                break;
                
}

```

**Comment by pacmano1** [comment, marked as answer]
```js
var dir = new java.io.File('/tmp');
var filter = new org.apache.commons.io.filefilter.WildcardFileFilter('*.csv')
var files = dir.list(filter)
for each(file in files)
{
                logger.info('filename ' + file);
}
```

**Comment by rutledj** [comment]
Well, that does put the filename out but with more info than I need.  I get this output:

`[2023-01-26 15:01:05.790]  INFO  (js-connector:?): ParameterizedMessage[messagePattern=filename CarePlan_2022-04-25T135050.000Z.csv, stringArgs=[], throwable=null]`

**Reply by pacmano1** [reply]
Ignore the logger.info funky formatting, that is a 4.1+ thing.   The point was the code, not the logger statement.

4.0.1
![image](https://user-images.githubusercontent.com/44065187/214941967-1d20384e-b621-4e6e-88d5-1dde05ddc9e8.png)

**Reply by pacmano1** [reply]
Please spend time reading other discussions, https://github.com/nextgenhealthcare/connect/discussions/5343

**Comment by rutledj** [comment]
Thx for your help. Is there a way to get rid of that extra data that is returned in the higher versions of Mirth (stringArgs=[], throwable=null)?

**Reply by pacmano1** [reply]
Don't use loggers except for catastrophic errors, rather store what you would be in a logger statement in the channelMap.   

Advantages:

1. The error/info/whatever statement in with its message
2. You don't clutter the console with crap.
